src/export/node_export_algorithm.rb

```diff
@@ -82,11 +87,98 @@ def run
 
   private
 
+  def offset_rotary_hinge_hubs(rotary_hinges, static_groups)
+    static_groups_as_sets_of_nodes = static_groups.map do |triangles|
+      nodes = Set.new
+      triangles.each do |triangle|
+        nodes << triangle.first_node
+        nodes << triangle.second_node
+        nodes << triangle.third_node
+      end
+      nodes
+    end
+
+    rotary_hinges.each do |node_pair|
+      static_groups_pair = static_groups_as_sets_of_nodes.select { |node_set| node_set.include?(node_pair[0]) && node_set.include?(node_pair[1]) }
```

Comment:
Might be worth replacing the node_pair data structure with an hash or [OpenStruct](https://ruby-doc.org/stdlib-2.5.1/libdoc/ostruct/rdoc/OpenStruct.html)? We could access the nodes more explicitly and increase readability then (something like `node_pair[:first]` and `node_pair[:second]`.

Reply:
If we stick to the array version we could simplify the check a bit, something like this:

```
...select { |node_set| (node_set.to_a & node_pair).any? }
```

Reply:
I'd stick with the array version, since it is possible that f.e. 4 nodes represent one rotation axis (then it's build a little bit redundant). If we later want to also handle that, the array would be nicer.